File: src/main/org/nlogo/prim/threed/_randomzcor.java
package org.nlogo.prim.threed;

import org.nlogo.agent.World3D;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Reporter;
import org.nlogo.nvm.Syntax;

public final strictfp class _randomzcor extends Reporter {
  @Override
  public Syntax syntax() {
    return Syntax.reporterSyntax(Syntax.TYPE_NUMBER);
  }

  @Override
  public Object report(Context context) {
    return report_1(context);
  }

  public double report_1(Context context) {
    double min = ((World3D) world).minPzcor() - 0.5;
    double max = ((World3D) world).maxPzcor() + 0.5;
    return min + context.job.random.nextDouble() * (max - min);
  }
}

File: src/main/org/nlogo/prim/threed/_randomzcor.scala
package org.nlogo.prim.threed

import org.nlogo.agent.World3D
import org.nlogo.api.Syntax
import org.nlogo.nvm.{ Context, Reporter }

class _randomzcor extends Reporter {
  override def syntax =
    Syntax.reporterSyntax(Syntax.NumberType)
  override def report(context: Context) = {
    val w = world.asInstanceOf[World3D]
    val min = w.minPzcor - 0.5
    val max = w.maxPzcor + 0.5
    java.lang.Double.valueOf(
      min + context.job.random.nextDouble() * (max - min))
  }
}
